roles/ceph-osd/tasks/standard.yml

```diff
@@ -1,4 +1,37 @@
 ---
+#  create ssd root bucket
+- name: Create root ssd-bucket root
+  shell: ceph osd crush add-bucket {{ ssd_pool }} root
+  when: inventory_hostname == groups['ceph_osds_ssd'][0]
+
+# Create crush ruleset for the ssd root bucket
+- name: Create crush ruleset ssd root bucket
+  shell: ceph osd crush rule create-simple {{ ssd_ruleset }} {{ ssd_pool }} host firstn
+  when: inventory_hostname == groups['ceph_osds_ssd'][0]
+
+# Create osd host bucket
+- name: Create osd host bucket
+  shell: ceph osd crush add-bucket {{ ansible_hostname }} host
+  register: result
+  changed_when: '"xx" in result.stdout'
+
+# Move the host bucket under corresponding root
+- name: Move host bucket under root bucket
+  shell: ceph osd crush move {{ ansible_hostname }} root={{ ssd_pool }}
```

Comment:
Are these tasks idempotent? can they be ran more than once and still get a 0 return code?